lib/dust.js

```diff
@@ -872,15 +873,21 @@
 
   Chunk.prototype.helper = function(name, context, bodies, params) {
     var chunk = this,
-        ret;
+        ret, filters;
     // handle invalid helpers, similar to invalid filters
     if(dust.helpers[name]) {
       try {
         ret = dust.helpers[name](chunk, context, bodies, params);
         if (dust.isThenable(ret)) {
-          return this.await(ret, context, bodies);
+          return chunk.await(ret, context, bodies);
+        }
+        if (ret instanceof Chunk) {
+          return ret;
+        }
+        if(typeof params.filters === 'string') {
+          filters = params.filters.split('|');
```

Comment:
I chose to ignore empty filters in `dust.filter` instead of messing with the string. If there's a better idiom I would be glad to switch.